python/_restclient/tests/test_restclient.py

```diff
@@ -150,3 +150,23 @@ def test_build_url(loop):
 
         assert client.build_url(base_url) == base_url
         assert client.build_url(base_url, query_params) == f"{base_url}?key=value"
+
+
+def test_restclient_nest_asyncio_ModuleNotFoundError(loop):
+    """Test for #99"""
+    import asyncio
+    import warnings
+
+    async def test():
+        await asyncio.sleep(0.01)
+
+        with warnings.catch_warnings():
+            # ignore coroutine not awaited warning for output sake.
+            # This is not the purpose of this test
+            warnings.simplefilter("ignore", category=RuntimeWarning)
+            with pytest.raises(ModuleNotFoundError):
+                # implicitly verify that `nest_asyncio` is not installed
+                # this test will need to change if `nest_asyncio` becomes a requirement
```

Comment:
Is it possible to elaborate in this comment a bit more concerning the implications of `nest_asyncio` becoming a requirement? I assume this test will fail is it's already installed. Would that require us to remove this test or write a new test? If `nest_asyncio` became a requirement, would that remove the need for the test? If so, why not just make it a requirement now?

Reply:
Yes you are correct -- if `nest_asyncio` is already installed that test will fail. My thinking was to treat this dependency differently since it isn't typically required. I agree with you about the fragility of this test, I need to rethink how to test this bug fix.